core/src/v2/models.rs

```diff
@@ -221,6 +223,73 @@ pub struct License {
     pub url: Option<String>,
 }
 
+/// Security Definition object.
+///
+/// https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#securityDefinitionsObject
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub struct SecurityDefinition {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<SecurityScheme>,
+}
+
+/// Security Scheme object.
+///
+/// https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#security-scheme-object
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub struct SecurityScheme {
+    pub type: String,
+    pub name: String,
+    pub in: String,
+    pub flow: String,
+    pub authorizationUrl: String,
+    pub tokenUrl: String,
+    pub scopes: Scopes,
```

Comment:
This should be `BTreeMap<String, String>`

Reply:
Could you please explain what `BTreeMap<String, String>` is doing?

Reply:
It's a collection of strings mapped to strings. That's exactly what the [scopes object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#scopesObject) demands. A field pattern `{name}` with type `string` means that for each string, there's a value of type `string`, which is strings mapped to strings.